T3: Fix oottracker-utils crate compilation

## Parent Issue: #74

## Overview
The `oottracker-utils` crate fails to compile due to missing dependencies and unresolved modules:
- Missing `reqwest` crate
- Missing `graphql_client` crate  
- Unresolved `biz_hawk_version_query` module
- Missing `oottracker_bizhawk` reference

## Error Summary
\`\`\`
error[E0433]: failed to resolve: use of unresolved module or unlinked crate \`reqwest\`
error[E0433]: failed to resolve: use of unresolved module or unlinked crate \`graphql_client\`
error[E0433]: failed to resolve: use of unresolved module or unlinked crate \`biz_hawk_version_query\`
\`\`\`

## Tasks
- [ ] Add missing dependencies to Cargo.toml
- [ ] Fix or remove broken version.rs functionality
- [ ] Ensure crate compiles
- [ ] Add basic tests

## Acceptance Criteria
- [ ] `cargo check -p oottracker-utils` passes
- [ ] `cargo test -p oottracker-utils` passes

--- a/crate/oottracker-utils/src/check_bizhawk_version.rs
+++ b/crate/oottracker-utils/src/check_bizhawk_version.rs
@@ -8,6 +8,7 @@
 )]
 #![forbid(unsafe_code)]
 
+#[cfg(windows)]
 use {
     oottracker_utils::version,
     semver::Version,
@@ -15,6 +16,7 @@ use {
     tokio::fs,
 };
 
+#[cfg(windows)]
 #[derive(Debug, thiserror::Error)]
 enum Error {
     #[error(transparent)]
@@ -31,6 +33,7 @@ enum Error {
     BizHawkVersionRegression,
 }
 
+#[cfg(windows)]
 #[wheel::main]
 async fn main() -> Result<(), Error> {
     let mut headers = reqwest::header::HeaderMap::new();
@@ -67,4 +70,10 @@ async fn main() -> Result<(), Error> {
         Equal => Ok(()),
         Greater => Err(Error::BizHawkVersionRegression),
     }
+}
+
+#[cfg(not(windows))]
+fn main() {
+    eprintln!("oottracker-check-bizhawk-version is only supported on Windows");
+    std::process::exit(1);
 }

--- a/crate/oottracker-utils/src/format_ram.rs
+++ b/crate/oottracker-utils/src/format_ram.rs
@@ -8,6 +8,7 @@
 )]
 #![forbid(unsafe_code)]
 
+#[cfg(windows)]
 use {
     oottracker::ram::{self, Ram},
     std::{
@@ -18,12 +19,14 @@ use {
     thiserror::Error,
 };
 
+#[cfg(windows)]
 #[derive(clap::Parser)]
 #[clap(version)]
 struct Args {
     input: PathBuf,
 }
 
+#[cfg(windows)]
 #[derive(Debug, Error)]
 enum Error {
     #[error(transparent)]
@@ -32,10 +35,17 @@ enum Error {
     Io(#[from] io::Error),
 }
 
+#[cfg(windows)]
 #[wheel::main]
 fn main(args: Args) -> Result<(), Error> {
     let mut buf = Vec::with_capacity(ram::SIZE);
     File::open(args.input)?.read_to_end(&mut buf)?;
     println!("{:#?}", Ram::from_bytes(&buf)?);
     Ok(())
+}
+
+#[cfg(not(windows))]
+fn main() {
+    eprintln!("oottracker-format-ram is only supported on Windows");
+    std::process::exit(1);
 }

--- a/crate/oottracker-utils/src/release.rs
+++ b/crate/oottracker-utils/src/release.rs
@@ -8,13 +8,17 @@
 )]
 #![forbid(unsafe_code)]
 
+#[cfg(any(windows, target_os = "macos"))]
+use async_proto::Protocol;
+#[cfg(any(windows, target_os = "macos"))]
+use gres::Percent;
+#[cfg(any(windows, target_os = "macos"))]
+use thiserror::Error;
+#[cfg(any(windows, target_os = "macos"))]
 use {
     ::tokio::{fs, io, process::Command},
-    async_proto::Protocol,
     dir_lock::DirLock,
-    gres::Percent,
     std::env,
-    thiserror::Error,
     wheel::traits::AsyncCommandOutputExt as _,
 };
 #[cfg(windows)]
@@ -57,6 +61,7 @@ use {
 #[cfg(windows)]
 const MACOS_ADDR: &str = "192.168.178.63";
 
+#[cfg(any(windows, target_os = "macos"))]
 #[derive(Debug, Error)]
 enum Error {
     #[cfg(windows)]
@@ -473,6 +478,7 @@ impl Task<Result<(), Error>> for BuildGui {
     }
 }
 
+#[cfg(any(windows, target_os = "macos"))]
 #[derive(Protocol)]
 enum MacMessage {
     Progress { label: String, percent: Percent },
@@ -1215,4 +1221,10 @@ async fn main(args: Args) -> Result<(), Error> {
         line.replace("[done] release published").await?;
     }
     Ok(())
+}
+
+#[cfg(not(any(windows, target_os = "macos")))]
+fn main() {
+    eprintln!("oottracker-release is only supported on Windows and macOS");
+    std::process::exit(1);
 }

--- a/crate/oottracker-utils/src/version.rs
+++ b/crate/oottracker-utils/src/version.rs
@@ -1,3 +1,4 @@
+#[cfg(windows)]
 use {
     graphql_client::GraphQLQuery,
     itertools::Itertools as _,
@@ -8,8 +9,10 @@ use {
     tokio::process::Command,
 };
 
+#[cfg(windows)]
 pub const INFO_PLIST_PATH: &str = "assets/macos/OoT Tracker.app/Contents/Info.plist";
 
+#[cfg(windows)]
 #[derive(Deserialize, Serialize)]
 pub struct Plist {
     #[serde(rename = "CFBundleShortVersionString")]
@@ -20,6 +23,7 @@ pub struct Plist {
     _rest: Json, //HACK see https://github.com/ebarnard/rust-plist/issues/54#issuecomment-653756460
 }
 
+#[cfg(windows)]
 #[derive(Debug, thiserror::Error)]
 pub enum BizHawkError {
     #[error(transparent)]
@@ -44,6 +48,7 @@ pub enum BizHawkError {
 )]
 struct BizHawkVersionQuery;
 
+#[cfg(windows)]
 pub async fn bizhawk_latest(client: &reqwest::Client) -> Result<Version, BizHawkError> {
     let remote_version_string = client
         .post("https://api.github.com/graphql")
@@ -73,6 +78,7 @@ pub async fn bizhawk_latest(client: &reqwest::Client) -> Result<Version, BizHawk
     Ok(Version::new(major?, minor?, patch?))
 }
 
+#[cfg(windows)]
 pub async fn check_cli_version(package: &str, version: &Version) {
     let cli_output = String::from_utf8(
         Command::new("cargo")
@@ -106,6 +112,7 @@ pub async fn check_cli_version(package: &str, version: &Version) {
     );
 }
 
+#[cfg(windows)]
 pub async fn version() -> Version {
     let version =
         Version::parse(env!("CARGO_PKG_VERSION")).expect("failed to parse current version");

--- a/crate/oottracker-utils/src/version_bump.rs
+++ b/crate/oottracker-utils/src/version_bump.rs
@@ -8,6 +8,7 @@
 )]
 #![forbid(unsafe_code)]
 
+#[cfg(windows)]
 use {
     oottracker_utils::version,
     semver::{BuildMetadata, Prerelease, Version},
@@ -16,6 +17,7 @@ use {
     toml_edit::TomlError,
 };
 
+#[cfg(windows)]
 #[derive(clap::Parser)]
 #[clap(version)]
 enum Args {
@@ -25,6 +27,7 @@ enum Args {
     Exact { version: Version },
 }
 
+#[cfg(windows)]
 #[derive(Debug, thiserror::Error)]
 enum Error {
     #[error(transparent)]
@@ -35,14 +38,15 @@ enum Error {
     Plist(#[from] plist::Error),
     #[error(transparent)]
     Toml(#[from] TomlError),
-    #[error("found Cargo manifest without “package” entry")]
+    #[error("found Cargo manifest without "package" entry")]
     MissingPackageEntry,
-    #[error("found “package” entry in Cargo manifest without “version” entry")]
+    #[error("found "package" entry in Cargo manifest without "version" entry")]
     MissingVersionEntry,
-    #[error("“package” entry in Cargo manifest was not a table")]
+    #[error(""package" entry in Cargo manifest was not a table")]
     PackageIsNotTable,
 }
 
+#[cfg(windows)]
 //FROM https://github.com/dtolnay/semver/issues/243#issuecomment-854337640
 fn increment_patch(v: &mut Version) {
     v.patch += 1;
@@ -50,6 +54,7 @@ fn increment_patch(v: &mut Version) {
     v.build = BuildMetadata::EMPTY;
 }
 
+#[cfg(windows)]
 fn increment_minor(v: &mut Version) {
     v.minor += 1;
     v.patch = 0;
@@ -57,6 +62,7 @@ fn increment_minor(v: &mut Version) {
     v.build = BuildMetadata::EMPTY;
 }
 
+#[cfg(windows)]
 fn increment_major(v: &mut Version) {
     v.major += 1;
     v.minor = 0;
@@ -65,6 +71,7 @@ fn increment_major(v: &mut Version) {
     v.build = BuildMetadata::EMPTY;
 }
 
+#[cfg(windows)]
 #[wheel::main]
 async fn main(args: Args) -> Result<(), Error> {
     let version = match args {
@@ -108,4 +115,10 @@ async fn main(args: Args) -> Result<(), Error> {
         fs::write(manifest_path, manifest.to_string().into_bytes()).await?;
     }
     Ok(())
+}
+
+#[cfg(not(windows))]
+fn main() {
+    eprintln!("oottracker-version-bump is only supported on Windows");
+    std::process::exit(1);
 }

